pkg/cmd/roachtest/test_runner.go

```diff
@@ -730,6 +730,11 @@ func (r *testRunner) runWorker(
 			return err
 		}
 
+		// Verify that the deprecated workload is available if needed.
+		if testToRun.spec.RequiresDeprecatedWorkload && workload[arch] == "" {
```

Comment:
Can we do this earlier? Maybe after test selection, if there is any test with RequiresDeprecatedWorkload, fail early or remove those from selected tests list with warning, as opposed to checking this for all the tests.

Reply:
I just copied the approach of how we verify that other binaries, i.e. `libgeos`, exist when the test marks them as required.

I agree that it'd be nice if it could know earlier and skip/fail them right at the start though. Let me look into it! 😃

Reply:
Ah, the reason it can't be earlier is because we need to know the arch of the test, which isn't picked until right above here: 

https://github.com/cockroachdb/cockroach/blob/f91b32003f394af57d7acbda3a6f0776045c4c40/pkg/cmd/roachtest/test_runner.go#L704-L721